model/project.go

```diff
@@ -322,22 +323,28 @@ func (c *PluginCommandConf) UnmarshalYAML(unmarshal func(interface{}) error) err
 // we maintain Params for backwards compatibility, but we read from YAML when available, as the
 // given params could be corrupted from the roundtrip
 func (c *PluginCommandConf) unmarshalParams() error {
-	if c.ParamsYAML != "" {
+	if c.ParamsYAML != "" && !isEmptyYAML(c.ParamsYAML) {
```

Comment:
This part of the approach makes sense to me. As an alternative, after you unmarshall `c.ParamsYAML` into `out`, could you check the length of `out`?

Reply:
The main reason I didn't want to do that is because if we ultimately want to get rid of Params, then we won't unmarshal ParamsYAML unless we need it, in which case I don't want this approach to rely on populating Params.